app/src/App.tsx

```diff
@@ -56,6 +57,28 @@ function App() {
   // An error message to display to the user.
   const [drawerOpen, setDrawerOpen] = useState(false);
 
+  useEffect(() => {
+    function setLocalStorageFromExternalDomain(message: MessageEvent) {
+      if (message.origin !== APPROVED_EXTERNAL_DOMAIN) {
+        return;
+      }
+
+      const data = JSON.parse(message.data);
+      const { swayCode } = data;
+      // I'm only expecting us to set sway code from the docs hub.
+      // If users want to see the corresponding solidity they can use the built-in transpiler.
+      if (swayCode) {
+        onSwayCodeChange(swayCode);
+      }
+    }
+
+    window.addEventListener('message', setLocalStorageFromExternalDomain);
```

Comment:
I tested this out, but it doesn't seem to be working. When the playground link opens, it doesn't have the code from the example, and I don't see any messages in the network tab. I'm guessing it could be a timing issue; perhaps the message is sent before the playground page is rendered (and this useEffect would run).

Reply:
yeah it's probably a timing issue.  The delay I added on the docshub was through trial and error.  Another thing is you need to run the docs hub with `NODE_ENV=development pnpm dev`